torchvision/models/detection/keypoint_rcnn.py

```diff
@@ -55,10 +57,10 @@ class KeypointRCNN(FasterRCNN):
             If box_predictor is specified, num_classes should be None.
         min_size (int): minimum size of the image to be rescaled before feeding it to the backbone
         max_size (int): maximum size of the image to be rescaled before feeding it to the backbone
-        image_mean (Tuple[float, float, float]): mean values used for input normalization.
+        image_mean (List[float, float, float]): mean values used for input normalization.
             They are generally the mean values of the dataset on which the backbone has been trained
             on
-        image_std (Tuple[float, float, float]): std values used for input normalization.
+        image_std (List[float, float, float]): std values used for input normalization.
```

Comment:
```suggestion
        image_std (List[float]): std values used for input normalization.
```

Reply:
Well, here is the catch. By annotating as `List[float]`.  The end-user will not know how many numbers to pass in.
Hence I propose to type with `Tuple[float, float, float]` and refactor code accordingly.

Also we can make using Tuple a Convention, where we know the number of elements.
E.g. case is this, where image_mean, image_std will be of length 3; the below case where image_size will be length 2.

Reply:
Hi!, once we get a consensus on using Tuple for finite element cases, I will refactor the code accordingly. :smile:

Reply:
> By annotating as `List[float]`. The end-user will not know how many numbers to pass in.

True, and this is why `List` is not a good type for this. Still, `List` only has a single inner type, so `List[float, float, float]` is simply a wrong annotation. If you use that inside the code instead of the docstring you will get an error.

> Also we can make using Tuple a Convention, where we know the number of elements.
E.g. case is this, where image_mean, image_std will be of length 3; the below case where image_size will be length 2.

I agree in general. Is this here such a case though? Are we somehow enforcing three channels or does the number of elements in `image_mean` and `image_std` just has to match the number of channels?

Reply:
I think we assume that channels are 3. We don't directly support any other channels without editing the backbone. So it might be safe assumption. I think there maybe a case where user has a backbone with different no of channels (say 2 or 4) and is trying to build FRCNN. Though I'm not very sure, as I haven't constructed a backbone with different channels.

Maybe Vasilas can chip in his thoughts, he knows it better cc @datumbox

Reply:
My 2 cents: It's true that out of the box we don't offer a method that builds a backbone with channels !=3. Nevertheless the API of the FRCNN does support this as one can pass an arbitrary backbone which could support arbitrary number of channels. I would really prefer it if we could use `Tuple[float, ...]` because that's the true type (immutable tuple of arbitrary many floats) but based on your latest findings that's not supported by JIT. So the next best thing that we got is `List[float]` as Philip suggested.